src/app/(simulation)/profil/_components/simulationAnswerList/_components/AnswerRow.tsx

```diff
@@ -5,26 +5,27 @@ import { useForm, useRule } from '@/publicodes-state'
 import { useRouter } from 'next/navigation'
 
 type AnswerProps = {
-  rule: string
+  ruleDottedName: string
   level: number
 }
 
-export default function Answer({ rule: dottedName, level }: AnswerProps) {
+export default function Answer({ ruleDottedName, level }: AnswerProps) {
   const { t } = useClientTranslation()
 
   const router = useRouter()
 
   const { setCurrentQuestion } = useForm()
 
-  const rule = useRule(dottedName)
+  const rule = useRule(ruleDottedName)
 
   const translateUnits = (units: string[]) => {
     return units.map((unit: string) => t(unit, { ns: 'units' }))
   }
 
-  const levelDottedName = dottedName.split(' . ')
+  const levelDottedName = ruleDottedName.split(' . ')
 
   let levelRule = undefined
+
   try {
     levelRule = useRule(
```

Comment:
Pas sur que ça se fasse les hooks dans des blocs try

Reply:
Ba écoute le eslint avec le plugin react hooks n'a pas gueulé, j'estime que c'est validé 😎

Reply:
Je me suis posé la question aussi, mais je n'ai pas noté de comportement bizarre ou de bug particulier, après le garde fou devrait sans doute plutôt être dans `useRule`

Reply:
Je vois qu'il y a un `safeEvaluate` dans `useRule` on peut donc se passer de ce try/catch

Reply:
Yes. Et a l'avenir partons du principe que toutes les verifications se font à la source dans publicodes-state. Sinon on va se retrouver à dupliquer des checks partout